devdoc/tqueue_requirements.md

```diff
@@ -212,6 +247,26 @@ TQUEUE_PUSH_RESULT TQUEUE_PUSH(T)(TQUEUE(T) tqueue, T* item, void* copy_item_fun
 
 - **SRS_TQUEUE_01_022: [** If the queue is full (current head >= current tail + queue size), `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`. **]**
 
+- If the queue is full (current head >= current tail + queue size):
+
+  - If the queue is not growable, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`.
+
+  - If the queue is growable:
+
+    - `TQUEUE_PUSH(T)` shall release in shared mode the lock used to guard the growing of the queue.
+
+    - `TQUEUE_PUSH(T)` shall acquire in exclusive mode the lock used to guard the growing of the queue.
+
+    - `TQUEUE_PUSH(T)` shall double the size of the queue.
+
+    - `TQUEUE_PUSH(T)` shall reallocate the array used to store the queue items based on the newly computed size.
+
+    - If reallocation fails, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`.
```

Comment:
>If reallocation fails, `TQUEUE_PUSH(T)` shall return `TQUEUE_PUSH_QUEUE_FULL`. [](http://example.com/codeflow?start=6&length=78)

I was wondering if we should return a different error indicating that this is a memory failure, not just too many item error.  /consider

Reply:
Agree, it is funny how this thought went through my head when I wrote it and then said "eeeeeh".

Reply:
Added _ERROR